### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false negatives for arbitrary substitution functions

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#8854]
### What minimal example or steps are needed to reproduce the bug?

```css
div {
	color: rgb(if(foo) 0 0);
	color: rgb(var(foo) 0 0);
	color: rgb(env(foo) 0 0);
	color: rgb(--foo(bar) 0 0);
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
	rules: {
    "declaration-property-value-no-unknown": true
  }
};
```

### How did you run Stylelint?

[Stylelint demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvCoQOYBGAFAJYBmVdEEAlCgAwcsDcW6ALviIly1AG5JCjZm07sefQRALFSlKjExjprDl16YBQ1aKoBaM0whUKk2fqwBfEABoQdGnhgA5JAFs4RBgAD38ABy8AOjAAZxjXcAhMDzIEEBCwon4UWDokAFc8bIxDfkJCmBiSEpwcdBBYMDxJJH4aJLMwwggwmEJ+AE8zCTx8mDNMCDN8zABrSYB3QwQUMrGsHEcnAwTIZJoyADEiP1a0mMGvPBpMfmiklMi-ACt4t1gw+MQalHqLgauN349RI9Wa-EqwNcG1+IH+gNuZj2KTMFyQmCgkigINh4Mh9ScIEcQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

I expected declarations containing arbitrary substitution functions to be validated according to the specification.

### What actually happened?

Some functions are always valid, even when they are used incorrectly (e.g. `var()`).
Others are always invalid.

### Do you have a proposal to fix the bug?

See: https://drafts.csswg.org/css-values-5/#resolve-property

> If a property value contains one or more [arbitrary substitution functions](https://drafts.csswg.org/css-values-5/#arbitrary-substitution-function), and all of those functions are themselves syntactically valid according to their [argument grammar](https://drafts.csswg.org/css-values-5/#argument-grammar)s, the entire value’s grammar must be assumed to be valid at parse time.

I propose that we check for the presence of known arbitrary substitution functions and then walk the AST.
In each AST we then only validate the arbitrary substitution functions themselves, and treat the declaration as valid of those parse correctly.

**Comment by romainmenke** [comment]
I thought this might work because I added the syntax for things like `if()` in `css-syntax-patches-for-csstree` but it seems that `parse()` in CSS Tree will still fail.

Even when `lexer(property, value)` (where `value` is a string) would pass and would use the syntax. I don't really understand the difference :)

----

I filed an upstream issue: https://github.com/csstree/csstree/issues/353

Closing this one, we can re-open if requested by users.
